src/mlt++/MltChain.cpp

```diff
@@ -27,10 +27,23 @@ Chain::Chain( ) :
 }
 
 Chain::Chain( Profile& profile, const char *id, const char *service ) :
-	instance( mlt_chain_init( profile.get_profile() ) )
+	instance( nullptr )
 {
-	Mlt::Producer source( profile, id, service );
-	mlt_chain_set_source( instance, source.get_producer() );
+	if ( id == NULL || service == NULL )
+	{
+		service = id != NULL ? id : service;
+		id = NULL;
+	}
+
+	mlt_producer source = mlt_factory_producer( profile.get_profile(), id, service );
+	if ( source )
+	{
+		instance = mlt_chain_init( profile.get_profile() );
+		mlt_chain_set_source( instance, source );
+		if ( id == NULL )
+			mlt_chain_attach_normalisers( instance );
```

Comment:
This might be worth a discussion. The loader producer does not return chains. So the normalizer link can not be added in the loader. I chose to add it here - mimicking the behavior of the loader. I am open to suggestions.

Reply:
If a chain could be sub-classed and added to repository and factor ala a producer, would there be a need for a any other chain besides a loader chain that contains this logic?

Reply:
A chain is a mechanism to connect links to a producer. So I can't think of a reason to make a subclass (except for the loader idea). Maybe the loader would be enough reason to make a registry for it.

Two other ideas:
* I could add a function to the factory which would add the normalizer logic - mlt_factory_chain(profile, service, resource)
* Somehow I could signal to the loader producer that it should encapsulate the producer in a chain. 

I will look into this some more.

Reply:
I vote to skip a loader chain and add mlt_factory_chain. I was thinking maybe to add an option to melt to ask it to add chains to the producers.

Reply:
In Melt, you can already add chains to producers by using the "-chain" option. But maybe you mean to add a chain to all producers without the need to specify the "-chain" option for each one?

Reply:
Oh yeah, maybe a global option later, but I think the normalizing links should be done (indirectly) by melt in addition to xml to mlt++

Reply:
I have most of this implemented (factory and ini file). However, I found an issue with this approach. If normalizing links are only added by the factory, then there is no way for an application to make a normalizing chain out of a producer that it already has created.

We do this in Shotcut here:
https://github.com/mltframework/shotcut/blob/287e3fae68c3509e3eddce934af91bcc3c9a3744/src/mltcontroller.cpp#L961

A couple of options:
1) Keep the mlt_chain_attach_normalisers() function so that applications can use it
2) Require the application to use the factory and then they will have to reapply the properties from the existing producer to the new chain
3) I could modify Shotcut so that all producers are always created as chains by the factory.

The advantage for #3 is that it would allow users to apply time remapping to image sequences and "xml as clips". Time remapping on "color" and "noise" clips probably does not make sense. But that probably doesn't matter. I am willing to do the work and debugging for this. But it would be a larger change with probably more  breakage than originally planned.

Reply:
1.) seems fine, and I suppose a corresponding `Mlt::Chain::attach_normalizers()` C++ method.
Please change spelling to be normal US English (z instead of s)? There is not any other symbols that use "normalise" that we need to be consistent. Charles Yates is Scottish and used all these British variations, and I have been slowly moving away from them (along with other abnormal things like code formatting).

Reply:
> The advantage for #3 is that it would allow users to apply time remapping to image sequences and "xml as clips". Time remapping on "color" and "noise" clips probably does not make sense. But that probably doesn't matter. I am willing to do the work and debugging for this. But it would be a larger change with probably more breakage than originally planned.

This is going to be very helpful when working with timelapses. Been wanting time remapping to images for a long time. Thanks!

Reply:
I have added Mlt::Chain::attach_normalizers()

I also changed many instances of "noralise" to "normalize". I changed as many as I could find that would not break backwards compatability.